bothan-api/server-cli/src/commands/start.rs

```diff
@@ -0,0 +1,196 @@
+use std::fs::{create_dir_all, File};
+use std::io::BufReader;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use anyhow::{anyhow, Context};
+use clap::Parser;
+use reqwest::header::{HeaderName, HeaderValue};
+use semver::VersionReq;
+use tonic::transport::Server;
+use tracing::{debug, info};
+
+use bothan_api::api::CryptoQueryServer;
+use bothan_api::config::ipfs::IpfsAuthentication;
+use bothan_api::config::manager::crypto_info::sources::CryptoSourceConfigs;
+use bothan_api::config::AppConfig;
+use bothan_api::proto::query::query_server::QueryServer;
+use bothan_api::VERSION;
+use bothan_binance::BinanceWorkerBuilder;
+use bothan_coingecko::CoinGeckoWorkerBuilder;
+use bothan_core::ipfs::{IpfsClient, IpfsClientBuilder};
+use bothan_core::manager::CryptoAssetInfoManager;
+use bothan_core::registry::{Registry, Valid};
+use bothan_core::store::SharedStore;
+use bothan_core::worker::AssetWorkerBuilder;
+
+#[derive(Parser)]
+pub struct StartCli {
+    /// The configuration file to use with bothan
+    #[arg(long)]
+    #[clap(default_value = "config.toml")]
+    config: PathBuf,
+
+    /// A flag to choose whether to start bothan as a fresh instance or not
+    #[arg(short, long)]
+    unsafe_reset: bool,
+
+    /// Flag to turn on dev mode
+    #[arg(short, long)]
+    dev: bool,
+
+    /// An optional initial registry state to use on startup to be used for testing purposes only.
+    #[arg(long, requires = "dev")]
+    registry: Option<PathBuf>,
+}
+
+impl StartCli {
+    pub async fn run(&self) -> anyhow::Result<()> {
+        let app_config = AppConfig::from(&self.config)?;
+
+        let registry = match &self.registry {
+            Some(p) => {
+                let file = File::open(p).map_err(|_| anyhow!("Failed to open registry file"))?;
+                let reader = BufReader::new(file);
+                serde_json::from_reader(reader)
+                    .map_err(|_| anyhow!("Failed to parse registry file"))?
+            }
+            None => Registry::default(),
+        };
+
+        let valid_registry = registry
+            .validate()
+            .map_err(|_| anyhow!("Invalid registry"))?;
```

Comment:
validate() should error with reason (can be custom error type or even string) and then can print it out to user